ddtrace/llmobs/experimentation/_dataset.py

```diff
@@ -0,0 +1,910 @@
+                    )
+                dataset_version = version
+            except ValueError as e:
+                if "404" in str(e):
+                    raise ValueError(f"Dataset '{name}' not found with version {version}") from e
+                raise
+
+        url = f"/api/unstable/llm-obs/v1/datasets/{dataset_id}/records"
+        if version is not None:
+            url += f"?filter[version]={version}"
+
+        try:
+            resp = exp_http_request("GET", url)
+            records_data = resp.json()
+        except ValueError as e:
+            if "404" in str(e):
+                if version is not None:
+                    versions_str = ", ".join(str(v) for v in sorted(available_versions))
+                    raise ValueError(
+                        f"Version {version} not found for dataset '{name}'. " f"Available versions: {versions_str}"
+                    ) from e
+                raise ValueError(f"Dataset '{name}' not found") from e
+            raise
+
+        # Transform records into the expected format
+        class_records = []
+        for record in records_data.get("data", []):
+            attrs = record.get("attributes", {})
+            input_data = attrs.get("input")
+            expected_output = attrs.get("expected_output")
+
+            class_records.append(
+                {
+                    "record_id": record.get("id"),
+                    "input": input_data,
+                    "expected_output": expected_output,
+                    **attrs.get("metadata", {}),
+                }
+            )
+
+        dataset = cls(name, class_records)
+        dataset._datadog_dataset_id = dataset_id
+        dataset._datadog_dataset_version = dataset_version
+        return dataset
+
+    def _batch_update(self, overwrite: bool = False) -> None:
+        """
+        Send all tracked changes to Datadog using the batch update endpoint.
+        This method is called by push() when there are pending changes.
+
+        Raises:
+            ValueError: If the dataset is not synced with Datadog (no dataset_id)
+            Exception: If the batch update request fails
+        """
+        if not self._datadog_dataset_id:
+            raise ValueError("Dataset must be synced with Datadog before performing batch updates")
+
+        if overwrite:
+            print(f"{Color.YELLOW}Warning: Overwriting existing dataset '{self.name}' using batch update{Color.RESET}")
+        else:
+            print("Pushing changes to existing dataset, new version will be created")
+
+        payload = self._prepare_batch_payload(overwrite=overwrite)
+
+        url = f"/api/unstable/llm-obs/v1/datasets/{self._datadog_dataset_id}/batch_update"
+        exp_http_request("POST", url, body=json.dumps(payload).encode("utf-8"))
+
+        # Sleep briefly to allow for processing on the backend using the constant
+        time.sleep(API_PROCESSING_TIME_SLEEP)
```

Comment:
can we return in the api a processing time sleep rather than hardcoding it in the client?